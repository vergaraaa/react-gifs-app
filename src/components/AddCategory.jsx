import React, { useState } from "react"

export const AddCategory = ({
  onAddCategory
}) => {
  const [search, setSearch] = useState("");

  const onInputChange = (e) => {
    setSearch(e.target.value)
  }

  const onSubmit = (e) => {
    e.preventDefault();

    if (!search.trim().length) return;

    onAddCategory(search)
    setSearch("");
  }

  return (
    <form onSubmit={onSubmit}>
      <input
        type="text"
        placeholder="Search gifs"
        value={search}
        onChange={onInputChange}
      />
    </form>
  )
}