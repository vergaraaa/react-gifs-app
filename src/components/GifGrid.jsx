import { GifGridItem } from "../components";
import { useGetGifs } from "../hooks/useGetGifs";

export const GifGrid = ({ category }) => {
  const { gifs, isLoading } = useGetGifs(category);

  return (
    <div>
      <h3>{category}</h3>

      {
        isLoading && <h2>Loading...</h2>
      }

      <div className="card-grid">
        {
          gifs.map(gif => (
            <GifGridItem key={gif.id} {...gif} />
          ))
        }
      </div>
    </div>
  )
}