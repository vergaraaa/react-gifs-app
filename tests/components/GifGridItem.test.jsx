import { render, screen } from "@testing-library/react"
import { GifGridItem } from "../../src/components/GifGridItem";

describe('Tests on GifGridItems.jsx', () => {
  const title = "Gif Grid Item";
  const url = "https://place-hold.it/300x500";

  test('should match snapshot', () => {
    const { container } = render(<GifGridItem title={title} url={url} />);

    expect(container).toMatchSnapshot();
  });

  test('should show image with correct url', () => {
    render(<GifGridItem title={title} url={url} />);

    const { src, alt } = screen.getByRole("img");

    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('should show title in component', () => {
    render(<GifGridItem title={title} url={url} />);

    expect(screen.getByText(title)).toBeTruthy();
  })
});