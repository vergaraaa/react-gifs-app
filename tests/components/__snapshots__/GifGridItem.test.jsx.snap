// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on GifGridItems.jsx should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Gif Grid Item"
      src="https://place-hold.it/300x500"
    />
    <p>
      Gif Grid Item
    </p>
  </div>
</div>
`;
